### PR TITLE
Jbaker/input review updates

### PR DESCRIPTION
**Pull request by jason-capsule42**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Review Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

<details>
<summary>
RC Checklist
</summary>

### Browsers

[Browsers Support Guide](https://auro.alaskaair.com/support/browsersSupport)

[Dev demo link](https://alaskaairlines.github.io/auro-formkit/)

Android

- [ ] Chrome
- [ ] Firefox

 iOS

- [ ] Chrome
- [ ] Firefox
- [ ] Safari

Desktop

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### Scenarios

- [ ] Validated linked issues with issue reporting team
- [ ] Test coverage report review

[Framework playground ](https://github.com/AlaskaAirlines/auro-framework-playground)

- [ ] Next React
- [ ] SvelteKit

</details><br>
**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

## Summary by Sourcery

Improve auro-input date formatting robustness, accessibility behavior, and localization examples.

New Features:
- Add new localization and ISO date format Storybook examples for auro-input.

Bug Fixes:
- Normalize date format tokens so uppercase format attributes correctly map to validation help text.
- Prevent cursor jumping in Safari for text-like inputs by restoring caret position after value updates.
- Ensure the password visibility toggle exposes its pressed state via aria-pressed for assistive technologies.
- Stabilize help-text DOM node identity across validity transitions so screen readers re-announce changes.
- Make ISO date formatting and date value formatting utilities return undefined for invalid or missing inputs instead of malformed strings or errors.

Enhancements:
- Consolidate and clarify BaseInput default property initialization for consistency and maintainability.

Documentation:
- Document default values for several auro-input properties and update date API examples to use ISO-formatted min/max values.

Tests:
- Add regression and edge-case tests covering help-text live region behavior, password toggle accessibility, format normalization, ISO/date utility error handling, and date formatting edge cases.

### CODE REVIEW

**Comment by jason-capsule42** [comment]
@sourcery-ai review

File: components/input/src/base-input.js [hunk 1]
```diff
@@ -1135,6 +1132,25 @@ export default class BaseInput extends AuroElement {
...
+    // formatter manages the cursor itself) doesn't get a competing write.
+    const { selectionStart } = this.inputElement;
+
+    if (this.setSelectionInputTypes.includes(this.type)) {
+      this.updateComplete.then(() => {
+        try {
+          this.inputElement.setSelectionRange(selectionStart, selectionStart);
+        } catch (error) { // eslint-disable-line
+          // Some input types (number/email in certain UAs) throw on
+          // setSelectionRange; swallow and let the native cursor stand.
```
Comment (rmenner) [inline components/input/src/base-input.js:1150]: What is the point of the try catch if the catch is commented out?

File: components/input/src/auro-input-util.js [hunk 2]
```diff
@@ -21,6 +21,15 @@ function formatISODate(isoStr, format) {
   try {
     const date = dateFormatter.stringToDateInstance(isoStr);
 
+    // `stringToDateInstance` returns an `Invalid Date` for malformed strings
+    // and `null` for non-string input — it does NOT throw. Without this
+    // guard, `getFullYear()`/`getMonth()`/`getDate()` on an Invalid Date
+    // yield `NaN`, and `String(NaN).padStart(4, '0')` produces literals
+    // like "0NaN/0NaN/0NaN" instead of `undefined`.
+    if (!(date instanceof Date) || Number.isNaN(date.getTime())) {
+      return undefined;
```
Comment (rmenner) [inline components/input/src/auro-input-util.js:30]: If this catches every error and returns undefined. A bug in the date code can look like normal bad input and never error out.

File: components/input/src/base-input.js [hunk 3]
```diff
@@ -706,6 +706,15 @@ export default class BaseInput extends AuroElement {
     }
     this.inputId = this.id ? `${this.id}-input` : window.crypto.randomUUID();
 
+    // Normalize the format token to lowercase so case-mixed values supplied
+    // via attribute (e.g. `format="MM/DD/YYYY"`) hit the `dateFormatMap`
+    // lookup inside `setCustomHelpTextMessage`. Without this, an uppercase
+    // format silently misses the map and leaves `setCustomValidityForType`
+    // unset.
+    if (this.format) {
+      this.format = this.format.toLowerCase();
```
Comment (rmenner) [inline components/input/src/base-input.js:715]: Not sure it's a good idea to override the user provided format.I feel like we should make a new normalized format value to use inside the component. Also I don't think this will prevent someone from dynamically setting it later in the life cycle. It should probably be moved to willUpdate or changedProperties.

File: components/input/src/base-input.js (hunk 1 above)
Comment (jason-capsule42) [reply components/input/src/base-input.js:1150]: the comment calls out that if our cursor positioning fails the catch just let's the browser default behavior execute instead.

File: components/input/src/auro-input-util.js (hunk 2 above)
Comment (jason-capsule42) [reply components/input/src/auro-input-util.js:30]: The only error that could happen here is when someone tried to pass a string to the date attribute that was not a valid date. What is the practical concern here?

File: components/input/src/base-input.js (hunk 3 above)
Comment (jason-capsule42) [reply components/input/src/base-input.js:715]: That would be a lot of code to refactor at this stage of the release prep. Open to this being logged as a new ticket for a future improvement.

File: components/input/src/auro-input-util.js (hunk 2 above)
Comment (rmenner) [reply components/input/src/auro-input-util.js:30]: It normalizes bad values to undefined. might be hard to catch errors

Comment (jason-capsule42) [reply components/input/src/auro-input-util.js:30]: If we don't do that it will actually break the JS. What do you propose we do instead?

Comment (rmenner) [reply components/input/src/auro-input-util.js:30]: We could narrow the check to just the documented cases (date === null for non-string input, isNaN(date.getTime()) for invalid date) so anything unexpected falls through instead of getting normalized. but fine to leave as is. I was just worried that it could cause issues later.

Comment (rmenner) [reply components/input/src/auro-input-util.js:30]: Never mind. Didn't realize this was an opt-in helper function

**Review by rmenner — APPROVED** [review]
